hashing/sha256.cpp

```diff
@@ -0,0 +1,227 @@
+/**
+ * @file
+ * @author [Md. Anisul Haque](https://github.com/mdanisulh)
+ * @brief Simple C++ implementation of the [SHA-256 Hashing Algorithm]
+ * (https://en.wikipedia.org/wiki/SHA-2)
+ *
+ * @details
+ * [SHA-2](https://en.wikipedia.org/wiki/SHA-2) is a set of cryptographic hash
+ * functions that was designed by the
+ * [NSA](https://en.wikipedia.org/wiki/National_Security_Agency) and first
+ * published in 2001. SHA-256 is a part of the SHA-2 family. SHA-256 is widely
+ * used for authenticating software packages and secure password hashing.
+ */
+
+#include <cassert>   /// For assert
+#include <cstdint>   /// For uint8_t, uint32_t and uint64_t data types
+#include <iostream>  /// For IO operations
+
+/**
+ * @namespace hashing
+ * @brief Hashing algorithms
+ */
+namespace hashing {
+/**
+ * @namespace SHA-256
+ * @brief Functions for the [SHA-256](https://en.wikipedia.org/wiki/SHA-2)
+ * algorithm implementation
+ */
+namespace sha256 {
+/**
+ * @brief Rotates the bits of a 32-bit unsigned integer
+ * @param n Integer to rotate
+ * @param rotate Number of bits to rotate
+ * @return uint32_t The rotated integer
+ */
+uint32_t rightRotate(uint32_t n, size_t rotate) {
+    return (n >> rotate) | (n << (32 - rotate));
+}
+
+/**
+ * @brief The SHA-256 algorithm
+ * @param input The input string to hash
+ * @return std::string The final hash value
+ */
+std::string sha256(std::string &input) {
```

Comment:
```suggestion
std::string sha256(const std::string &input) {
```
`input` is not modified, so it should be `const`.